src/lang-go.ts

```diff
@@ -727,6 +733,94 @@ export async function activateUsingWebSockets(ctx: sourcegraph.ExtensionContext)
+    ctx.subscriptions.add(
+        registerFeedbackButton({
+            languageID: 'go',
+            sourcegraph,
+            isPrecise: false,
+        })
+    )
+
+    activateBasicCodeIntel({
+        sourcegraph,
+        languageID: 'go',
+        fileExts: ['go'],
+        filterDefinitions: ({ repo, filePath, pos, fileContent, results }) => {
+            const currentFileImportedPaths = fileContent
+                .split('\n')
+                .map(line => {
+                    // Matches the import at index 3
+                    const match = /^(import |\t)(\w+ |\. )?"(.*)"$/.exec(line)
+                    return match ? match[3] : undefined
+                })
+                .filter((x): x is string => Boolean(x))
+
+            const currentFileImportPath = repo + '/' + path.dirname(filePath)
+
+            const filteredResults = results.filter(result => {
+                const resultImportPath = result.repo + '/' + path.dirname(result.file)
+                return (
+                    currentFileImportedPaths.some(i => resultImportPath.includes(i)) ||
+                    resultImportPath === currentFileImportPath
+                )
+            })
+
+            return filteredResults.length === 0 ? results : filteredResults
+        },
+        commentStyle: {
+            lineRegex: /\/\/\s?/,
+        },
+    })(ctx)
+}
+
+async function activatePreciseUsingLspClient(ctx: sourcegraph.ExtensionContext): Promise<void> {
+    ctx.subscriptions.add(registerFeedbackButton({ languageID: 'go', sourcegraph, isPrecise: true }))
+
+    const token = await getOrTryToCreateAccessToken()
+
+    const serverURL = sourcegraph.configuration.get<Settings>().value['go.serverUrl']
+    const client = await lspClient.register({
+        sourcegraph,
+        transport: lspClient.webSocketTransport({
+            serverUrl: serverURL!,
+            logger: new NoopLogger(),
+        }),
+        documentSelector: [{ language: 'go' }],
+        initializationOptions: {
+            zipURLTemplate: zipURLTemplate(token),
+        },
+    })
+    ctx.subscriptions.add(client)
+
+    const settings: BehaviorSubject<Settings> = new BehaviorSubject<Settings>({})
+    ctx.subscriptions.add(
+        sourcegraph.configuration.subscribe(() => {
+            settings.next(sourcegraph.configuration.get<Settings>().value)
+        })
+    )
+
+    ctx.subscriptions.add(
+        registerWhile({
+            register: () =>
+                sourcegraph.languages.registerReferenceProvider([{ pattern: '*.go' }], {
+                    provideReferences: (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) =>
+                        xrefs({
+                            doc,
+                            pos,
+                            sendRequest: ({ rootURI, requestType, request, useCache }) =>
+                                client.withConnection(rootURI, conn => conn.sendRequest(requestType, request)),
```

Comment:
This is the part that gets into an infinite loop without https://github.com/sourcegraph/lsp-client/pull/42